libbeat/beat/beat.go

```diff
@@ -84,7 +84,7 @@ type Beat struct {
 // BeatConfig struct contains the basic configuration of every beat
 type BeatConfig struct {
 	// output/publishing related configurations
-	Output config.Namespace `config:"output"`
+	Output config.Namespace `config:"outputs"` // this seems to be the issue. But right now it seems that somehow it works when loading the config, but later on Output is empty again
```

Comment:
This was also the source of the problem in https://github.com/elastic/elastic-agent/issues/1860#issuecomment-1335794046

The Beats assume they are reading a complete configuration file at startup to populate that. This is not true for the V2 agent. There is no output configured at startup.

In V1 we actually did read the reference Filebeat configuration at startup, and this worked by coincidence because the default output is Elasticsearch.

Ideally what we do in V2 is correct, don't assume an output until the agent configures it. This avoids problems where the Beat loads something from the reference config that conflicts with what the agent sent it, or problems where the user notices the Beats are reaching out to localhost:9200 at startup when a Logstash output is configured.

The problem is the Beats were written assuming they could read a valid config file at startup, so this is causing some weird problems in code that assumes this is populated when it isn't necessarily.

The original issue for this change was https://github.com/elastic/beats/issues/31901

Reply:
This is where a blank default configuration is created: https://github.com/elastic/beats/blob/d8204a2c9f76c5abecca3af77136cea4901d0ae6/libbeat/cfgfile/cfgfile.go#L166

Reply:
If the problem is that the ingest pipeline setup is being skipped, that should not actually be the cause of this problem. See https://github.com/elastic/beats/pull/34143/files#r1059158293.